be/src/exprs/vectorized/array_functions.h

```diff
@@ -40,6 +40,27 @@ class ArrayFunctions {
 
 #undef DEFINE_ARRAY_DISTINCT_FN
 
+#define DEFINE_ARRAY_OVERLAP_FN(NAME, PT)                                                     \
+    static ColumnPtr array_overlap_##NAME(FunctionContext* context, const Columns& columns) { \
+        return ArrayOverlap<PT>::process(context, columns);                                   \
+    }
+
+    DEFINE_ARRAY_OVERLAP_FN(boolean, PrimitiveType::TYPE_BOOLEAN);
```

Comment:
the type enumeration code could be replaced with `olap_type_infra.h`

Reply:
> the type enumeration code could be replaced with `olap_type_infra.h`

Micro in olap_type_infra.h is dispatch with FieldType, codes here is dispatch with PrimitiveType. could I add this micro to olap_type_infra.h?